### PR TITLE
ELPP-3219 Make Button Collection a proper pattern

### PR DESCRIPTION
**Pull request by thewilkybarkid**
This is the long-way round getting a margin underneath buttons when they're part of the body content.

For future use (imagining buttons side by side), I decided to create a pattern that wrapped one or more buttons then add the margin to that, rather than try and add it to some buttons and not others (eg in the site header). Turns out that the Metrics pattern already had an informal pattern called Button Collection, so I've sorted that out.

I've also renamed the `button--outline` variant to `button--reversed`, then created a new variant called `button--outline` which is just an outline (not reversed as well!).

### CODE REVIEW

File: assets/sass/patterns/molecules/button-collection.scss [hunk 1]
```diff
@@ -2,42 +2,37 @@
...
-.button-collection__button--active {
-  background-color: $color-primary;
-  color: white;
 
-  &:hover {
-    background-color: $color-primary;
-    color: white;
+  .button-collection--compact & {
+    &:not(:first-child) {
+      .button {
```
Comment (thewilkybarkid) [inline assets/sass/patterns/molecules/button-collection.scss]: Getting pretty specific...

File: assets/sass/patterns/molecules/button-collection.scss [hunk 2]
```diff
@@ -2,42 +2,37 @@
 
 .button-collection {
   display: table;
```
Comment (davidcmoulton) [inline assets/sass/patterns/molecules/button-collection.scss]: As this is being updated anyway, could use flexbox here?

File: assets/sass/patterns/molecules/carousel-item.scss [hunk 3]
```diff
@@ -214,6 +214,9 @@
     bottom: 64px;
     left: 0;
     right: 0;
+    .button {
+      @include margin(0, "bottom");
```
[the remark this answers is not in the document]
Comment (davidcmoulton) [reply assets/sass/patterns/molecules/carousel-item.scss]: Correction: `margin-bottom: 0;`

**Comment by davidcmoulton** [comment]
👍

**Pull request merged**